build-tools/packages/build-cli/src/commands/setReleaseTagForPublishing.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Flags } from "@oclif/core";
+import { Package, updatePackageJsonFile } from "@fluidframework/build-tools";
+import path from "node:path";
+import { PackageCommand } from "../BasePackageCommand";
+import * as fs from "fs";
+
+/**
+ * A type defining the four release types
+ */
+type UpdatePackageJsonType = "alpha" | "beta" | "public";
+
+/**
+ * A typeguard to check if a string is a {@link UpdatePackageJsonEnum}.
+ */
+function isUpdatePackageJsonTypes(type: string): type is UpdatePackageJsonType {
+	return type === "alpha" || type === "beta" || type === "public";
```

Comment:
You can define the set of accepted values in a const array and then

See here for an example: https://github.com/microsoft/FluidFramework/blob/8ffc7274eb9a4678cc3ba5df028eaf939928864c/build-tools/packages/build-cli/src/releaseGroups.ts#L18-L36

(if you remove the `| string` from the ReleaseGroup type it is just the same as `type ReleaseGroup = "build-tools" | "client" | "server" | "gitrest" | "historian"`)

Then your isUpdatePackageJsonStypes type guard can call `.includes` on the array.